g_docs.go

```diff
@@ -904,3 +822,327 @@ func urlReplace(src string) string {
+	if !strings.Contains(pkgObject, ".") {
+		*realTypes = append(*realTypes, pkgObject)
+		return
+	}
+
+	pkgObjectSplit := strings.Split(pkgObject, ".")
+
+	if len(pkgObjectSplit) != 2 {
+		ColorLog("[WARN] %v pkgObject passed to realTypes length should be 2", pkgObjectSplit)
+		return
+	}
+
+	pkg := pkgObjectSplit[0]
+	object := pkgObjectSplit[1]
+
+	realType := object
+	if v, ok := pathInfo[pkg]; ok && v != "" {
+		realType = v + "." + object
+	}
+
+	realType = strings.Trim(realType, "/")
+	realType = strings.Replace(realType, "/", ".", -1)
+	*realTypes = append(*realTypes, realType)
+}
+
+// deepCopy copies all fields in an interface recursively.
+func deepCopy(value interface{}) (interface{}, error) {
+
+	// deep copy value of a nil is still a nil with a different pointer.
+	if value == nil {
+		return nil, nil
+	}
+
+	data, err := json.Marshal(value)
+	if err != nil {
+		return nil, err
+	}
+
+	vPtr := reflect.New(reflect.TypeOf(value))
+	err = json.Unmarshal(data, vPtr.Interface())
+	if err != nil {
+		return nil, err
+	}
+	return vPtr.Elem().Interface(), nil
+}
+
+// replicateSwaggerItem replicates a *swagger.Item struct using deepcopy then
+// casting it back to *swagger.Item.
+func replicateSwaggerItem(item *swagger.Item) (*swagger.Item, error) {
+	copiedItem, err := deepCopy(item)
+	if err != nil {
+		return nil, err
+	}
+
+	swaggerItem, ok := copiedItem.(*swagger.Item)
+	if !ok {
+		return nil, fmt.Errorf("Failed to cast item to *swagger.Item")
+	}
+
+	return swaggerItem, nil
+}
+
+// enrichItem enriches methods listed in *swagger.Item with tag and route data.
+func enrichItem(item *swagger.Item, tag, route string) {
+	v := reflect.ValueOf(item).Elem()
+	fields := make([]interface{}, v.NumField())
+
+	// Iterate through all `methods` in *swagger.Item,
+	// append tag and endpoint route to make all operationIDs unique
+	// methods: Get, Put, Post, etc.
+	for i := 0; i < v.NumField(); i++ {
+		fields[i] = v.Field(i).Interface()
+
+		// Skip non-method fields.
+		method, ok := fields[i].(*swagger.Operation)
+		if !ok {
+			continue
+		}
+
+		// current method is not used by the endpoint.
+		if method == nil {
+			continue
+		}
+
+		method.Tags = append(method.Tags, tag)
+
+		// OperationID is optional and can be missing if @Title is not set
+		if method.OperationID != "" {
+			method.OperationID = operationIDFormat(route) + "." + method.OperationID
+		}
+	}
+}
+
+// operationIDFormat transforms a string into a general format used in OperationID
+func operationIDFormat(value string) string {
+	value = strings.Trim(value, "/")
+	value = strings.Replace(value, "/", ".", -1)
+
+	return value
+}
+
+// fieldNameFromTag process a tag attached to a struct field to get
+// the respective name according to its encoding (thrift/json).
+// if the tag is explicityly wants to be ignored then
+// the name returned will be an empty string.
+func fieldNameFromTag(tag string) string {
+	var name string
```

Comment:
A variable should be declared as close where it is used - line 1123 in this case.